### ISSUE TITLE
C# properties throwing Ref<> errors in 2.0

### ISSUE THREAD

**Issue by Wenzil** [jacobdufault/fullinspector#31]
C# properties still throw Ref<> errors for non-`BaseBehavior` component types. For example, properties of type `RigidBody` or any custom `MonoBehaviour` still throw errors. Weirdly enough, `Transform` properties are ok. 

So in this code, everything works fine except that `SomeBehavior.body` and `SomeBehavior.normalBehavior` cause the following errors to show in the console: 

```
A property of type UnityEngine.Rigidbody needs to be wrapped in Ref<>
UnityEngine.Debug:LogError(Object)
```

```
A property of type NormalBehavior needs to be wrapped in Ref<>
UnityEngine.Debug:LogError(Object)
```

`SomeBehavior.target` and `SomeBehavior.anotherBehavior` don't cause any error. 

Note that replacing C# properties with fields of the same type gets rid of any error. Note also that all four `Debug.Log()` successfully print out their injected instance, regardless of errors being thrown.

Here's the code:

``` csharp
public class NormalBehavior : MonoBehaviour
{
    public Vector3 someVector;
}

public class AnotherBehavior : BaseBehavior
{
    [SerializeField]
    [ShowInInspector]
    private int someInt { get; set; }

    [SerializeField]
    [ShowInInspector]
    private float someFloat { get; set; }
}
```

``` csharp
public class SomeBehavior : BaseBehavior
{
    [SerializeField]
    [ShowInInspector]
    private Transform target { get; set; }

    [SerializeField]
    [ShowInInspector]
    private Rigidbody body { get; set; }

    [SerializeField]
    [ShowInInspector]
    private NormalBehavior normalBehavior { get; set; }

    [SerializeField]
    [ShowInInspector]
    private AnotherBehavior anotherBehavior { get; set; }

    void Start()
    {
        Debug.Log(target);
        Debug.Log(body);
        Debug.Log(normalBehavior);
        Debug.Log(anotherBehavior);
    }
}
```
